pickle-plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt

```diff
@@ -121,39 +119,28 @@ class PicklePlugin : Plugin<Project> {
     }
 
     private fun configureAndroidTest(project: Project, variants: DomainObjectSet<out TestVariant>, extension: Extension) {
-        val featuresDir = extension.androidTest.featuresDir ?: throw IllegalStateException("You must specify \"featuresDir\" inside \"androidTest\" for pickle to work with Android tests")
-
-        configure(project, variants, extension, featuresDir, { variant, dir -> File(variant.resolveOutputDir(), dir) }, { task, variant -> task.setupDependency(variant) })
-    }
+        val featuresDir = extension.androidTest.featuresDir
+            ?: throw IllegalStateException("You must specify \"featuresDir\" inside \"androidTest\" for pickle to work with Android tests")
 
-    /**
-     * Setup Pickle task dependency in a backward compatible way.
-     *
-     * mergeAssetsProvider was only introduced by Android Gradle Plugin 3.3.
-     *
-     * Remove this when we want to only support Android Gradle Plugin 3.3 and above.
-     */
-    private fun GenerateTask.setupDependency(variant: BaseVariant) {
-        try {
-            dependsOn(variant.mergeAssetsProvider)
-        } catch (ignored: Throwable) {
-            dependsOn(variant.mergeAssets)
+        configure(project, variants, extension) { variant, directoryProperty ->
+            directoryProperty.set(variant.mergeAssetsProvider.flatMap { it.outputDir.dir(featuresDir) })
```

Comment:
Doing this `flatMap` here signals Gradle that we need this particular input to proceed. Not only this helps with Task caching etc, but also setups correct dependencies between tasks.